### PR TITLE
feat: feature search basic functionality

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Basic search

### CODE REVIEW

File: src/lib/openapi/spec/feature-search-query-parameters.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { FromQueryParams } from '../util/from-query-params';
+
+export const featureSearchQueryParameters = [
+    {
+        name: 'query',
+        schema: {
+            default: '',
```
Comment (kwasniew) [inline src/lib/openapi/spec/feature-search-query-parameters.ts:7]: we need to investigate why the default is not set

File: src/lib/openapi/spec/feature-search-query-parameters.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            type: 'string' as const,
+            example: 'feature_a',
+        },
+        description: 'The search query for the feature or tag',
+        in: 'query',
+    },
+    {
+        name: 'projectId',
+        schema: {
+            default: '',
```
Comment (kwasniew) [inline src/lib/openapi/spec/feature-search-query-parameters.ts:17]: store should do * or %

**Review by FredrikOseberg — APPROVED** [review]
LGTM

File: src/lib/features/feature-search/feature-search-controller.ts [hunk 3]
```diff
@@ -56,6 +55,8 @@ export default class FeatureSearchController extends Controller {
                     summary: 'Search and filter features',
                     description: 'Search and filter by selected fields.',
                     operationId: 'searchFeatures',
+                    // TODO: fix the type
+                    parameters: featureSearchQueryParameters as any,
```
Comment (kwasniew) [inline src/lib/features/feature-search/feature-search-controller.ts:59]: @thomasheartman can you help here? Do we have a standard method for externalizing query params to a type without type assertions?

**Pull request merged**